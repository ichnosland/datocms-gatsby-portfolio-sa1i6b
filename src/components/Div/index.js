/**
*
* Div
*
*/

import styled, { css } from 'styled-components';
import media from 'style/mediainjector';

const Div = styled.div`
  display: ${(props) => props.display ? props.display : 'block'};
  position: ${(props) => props.position ? props.position : 'relative'};
  width: ${(props) => props.width};
  max-width: ${(props) => props.maxWidth ? props.maxWidth : '100%'};
  height: ${(props) => props.height};
  color: ${(props) => props.color ? props.color : 'inherit'};
  text-align: ${(props) => props.align};
  float: ${(props) => props.float ? props.float : 'none'};
  margin: ${(props) => props.margin ? props.margin : '0'};
  padding: ${(props) => props.padding ? props.padding : '0'};
  border: ${(props) => props.border ? props.border : 'none'};
  border-width: ${(props) => props.borderwidth};
  border-style: ${(props) => props.borderstyle};
  border-color: ${(props) => props.bordercolor};
  background-color: ${(props) => props.bgColor ? props.bgColor : 'transparent'};
  border-radius: ${(props) => props.radius};
  overflow: ${(props) => props.overflow ? props.overflow : 'visible'};
  user-select: ${(props) => props.userSelect};
  pointer-events: ${(props) => props.pointerEvents};
  z-index: ${(props) => props.zindex};
  ${(props) => props.nobreak && css`
    page-break-inside: avoid;
  `}
  ${media.print`
    ${({ noPrint }) => noPrint && css`
        display:none;
    `}
  `}
`;

export default Div;
