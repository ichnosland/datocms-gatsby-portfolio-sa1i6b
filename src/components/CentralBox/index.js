/**
*
* CentralBox
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import Div from '../Div';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const FramedBox = styled(Div)`
  position: relative;
  width: 100%;
  padding: 95px 30px 100px;
  ${media.desktop`
    min-height: 480px;
    margin: 20px auto;
    background-color: ${(props) => props.unita ? props.theme.brand : colore.ui.contrast};
    border-radius: ${(props) => props.versione ? '3px' : '20px 20px 48px 48px'};
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.3);
    ${(props) => props.unita && css`
      padding-bottom: 30px;
    `};
  `};
  ${(props) => props.unita && css`
    display: flex;
    flex-direction: column;
    justify-content: center;
    min-height: 0;
    padding: 66px 30px 30px;
    color: ${colore.ui.contrast};
    text-align: center;
    background-color: ${props.theme.brand};
    h2 {
        padding: 0 30px 40px;
        ${media.lt667`
            padding: 0 20px 60px;
        `};
      }
    ${media.mobile`
      height: 100%;
      overflow: auto;
    `};
    ${media.lt480`
      padding: 66px 10px 30px;
    `};
    ${media.iphone45`
      justify-content: flex-start;
      h2 {
        padding: 10px 0 20px;
      }
    `};
  `};
  ${media.mobile`
    height: 100%;
  `};
  * {
    user-select: none;
  }
`;

FramedBox.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

class CentralBox extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <FramedBox
        unita={this.props.unita}
        versione={this.props.versione}
      >
        {this.props.children}
      </FramedBox>
    );
  }
}

CentralBox.propTypes = {
  unita: PropTypes.bool,
  versione: PropTypes.bool,
  children: PropTypes.any,
};

export default CentralBox;
