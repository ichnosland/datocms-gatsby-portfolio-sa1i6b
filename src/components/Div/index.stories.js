import React from 'react';
import { storiesOf } from '@storybook/react';
import { withNotes } from '@storybook/addon-notes';
import { withKnobs, text, boolean, color } from '@storybook/addon-knobs';
import Div from './index';


const nota = 'La props nobreak assegna \'page-break-inside: avoid;\' \n quella noPrint \'display:none;\' in stampa';

storiesOf('Div', module)
  .addDecorator(withKnobs)
  .addWithJSX('Div', withNotes(nota)(() => (
    <Div
      display={text('display', 'block')}
      position={text('position', 'relative')}
      width={text('width', 'auto')}
      maxWidth={text('maxWidth', '100%')}
      height={text('height', '')}
      color={color('color', 'inherit')}
      align={text('align (tex-align)', '')}
      float={text('float', 'none')}
      margin={text('margin', '0')}
      padding={text('padding', '0')}
      radius={text('radius', '')}
      bgColor={color('bgColor', 'transparent')}
      border={text('border', '')}
      borderwidth={text('borderwidth', '')}
      borderstyle={text('borderstyle', '')}
      bordercolor={color('bordercolor', '')}
      overflow={text('overflow', 'visible')}
      userSelect={text('userSelect', 'auto')}
      pointerEvents={text('pointerEvents', 'auto')}
      zindex={text('zindex', 'auto')}
      nobreak={boolean('nobreak', false)}
      noPrint={boolean('noPrint', true)}
    >
      <p>Ciao, sono all&apos;interno di un Div. Usa le props per personalizzarlo :)</p>
    </Div>
  )));
