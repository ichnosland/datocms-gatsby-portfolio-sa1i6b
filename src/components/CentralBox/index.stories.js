import React from 'react';
import { Router } from 'react-router-dom';
import createBrowserHistory from 'history/createBrowserHistory';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';
import { ThemeProvider } from 'styled-components';
import { allThemes, themeOptions } from 'style/theme';
import TopBar from 'components/TopBar';
import { ContainerFlex } from 'components/Container';
import ProgressBar from 'components/ProgressBar';
import ToolBar from 'components/ToolBar';
import ToolBarButton from 'components/ToolBar/ToolBarButton';
import CentralBox from './index';

const history = createBrowserHistory();

storiesOf('CentralBox', module)
  .addDecorator(withKnobs)
  .add('CentralBox', () => (
    <ThemeProvider theme={allThemes[select('theme', themeOptions, 'alatin')]}>
      <ContainerFlex height="100%">
        <CentralBox
          unita={boolean('unita', false)}
          versione={boolean('versione', false)}
        >
          {text('Text', 'Testo a caso :)')}
        </CentralBox>
      </ContainerFlex>
    </ThemeProvider>
  ))
  .add('CentralBox popolata', () => (
    <ThemeProvider theme={allThemes[select('theme', themeOptions, 'alatin')]}>
      <ContainerFlex height="100%">
        <CentralBox
          versione={boolean('versione', false)}
        >
          <ProgressBar
            lezione
            percentuale={25}
          />
          <Router history={history}>
            <TopBar
              esecuzione
              noShadow
              box
              closeBtn={{
                url: '#',
                enabled: true,
              }}
              lesson={{
                url: '#',
                enabled: true,
              }}
              score="+ 120"
            />
          </Router>
          {text('Text', 'Testo a caso :)')}
          <ToolBar>
            <ToolBarButton
              full
              title="Controlla"
            />
          </ToolBar>
        </CentralBox>
      </ContainerFlex>
    </ThemeProvider>
  ));

