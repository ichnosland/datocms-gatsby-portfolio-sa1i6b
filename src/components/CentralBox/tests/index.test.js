import React from 'react';
import { shallow } from 'enzyme';

import Div from '../Div';
import CentralBox, { FramedBox } from '../index';

const children = <p>some text</p>;

describe('<CentralBox />', () => {
  it('should render a FramedBox tag', () => {
    const renderedComponent = shallow(<CentralBox />);
    expect(renderedComponent.find(FramedBox).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its childrens', () => {
    const renderedComponent = shallow(<CentralBox>{children}</CentralBox>);
    expect(renderedComponent.contains(children)).toBe(true);
  });
});

describe('<FramedBox />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(<FramedBox />);
    expect(renderedComponent.find(Div).length).toEqual(1);
  });

  it('should render its css w/ props unita', () => {
    const renderedComponent = shallow(<FramedBox unita />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props versione', () => {
    const versione = '0';
    const renderedComponent = shallow(<FramedBox versione={versione} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props theme', () => {
    const theme = {
      brand: '#00abe5',
    };
    const renderedComponent = shallow(<FramedBox theme={theme} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
