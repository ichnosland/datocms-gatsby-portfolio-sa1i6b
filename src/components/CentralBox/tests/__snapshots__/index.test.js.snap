// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CentralBox /> should render a FramedBox tag 1`] = `
<CentralBox__FramedBox
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;

exports[`<FramedBox /> should render its css w/ props theme 1`] = `
.c0 {
  position: relative;
  width: 100%;
  padding: 95px 30px 100px;
}

.c0 * {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

@media (min-width:64.0625em) {
  .c0 {
    min-height: 480px;
    margin: 20px auto;
    background-color: rgb(255,255,255);
    border-radius: 20px 20px 48px 48px;
    box-shadow: 0 1px 2px rgba(0,0,0,0.3);
  }
}

@media (max-width:64em) {
  .c0 {
    height: 100%;
  }
}

<Div
  className="c0"
  theme={
    Object {
      "brand": "#00abe5",
    }
  }
/>
`;

exports[`<FramedBox /> should render its css w/ props unita 1`] = `
.c0 {
  position: relative;
  width: 100%;
  padding: 95px 30px 100px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 0;
  padding: 66px 30px 30px;
  color: rgb(255,255,255);
  text-align: center;
  background-color: #00BBEF;
}

.c0 h2 {
  padding: 0 30px 40px;
}

.c0 * {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

@media (min-width:64.0625em) {
  .c0 {
    min-height: 480px;
    margin: 20px auto;
    background-color: #00BBEF;
    border-radius: 20px 20px 48px 48px;
    box-shadow: 0 1px 2px rgba(0,0,0,0.3);
    padding-bottom: 30px;
  }
}

@media (max-width:41.6875em) {
  .c0 h2 {
    padding: 0 20px 60px;
  }
}

@media (max-width:64em) {
  .c0 {
    height: 100%;
    overflow: auto;
  }
}

@media (max-width:30em) {
  .c0 {
    padding: 66px 10px 30px;
  }
}

@media (max-device-width:20em) {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }

  .c0 h2 {
    padding: 10px 0 20px;
  }
}

@media (max-width:64em) {
  .c0 {
    height: 100%;
  }
}

<Div
  className="c0"
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
  unita={true}
/>
`;

exports[`<FramedBox /> should render its css w/ props versione 1`] = `
.c0 {
  position: relative;
  width: 100%;
  padding: 95px 30px 100px;
}

.c0 * {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

@media (min-width:64.0625em) {
  .c0 {
    min-height: 480px;
    margin: 20px auto;
    background-color: rgb(255,255,255);
    border-radius: 3px;
    box-shadow: 0 1px 2px rgba(0,0,0,0.3);
  }
}

@media (max-width:64em) {
  .c0 {
    height: 100%;
  }
}

<Div
  className="c0"
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
  versione="0"
/>
`;
